src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,109 @@
 package co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
+
+public class Range<T extends Comperable> implements Set<T> {
+
+    private ArrayList<T> elementData  = new ArrayList<>();
+    private T to;
+    private T from;
+
+    public Range(T from, T to, Function<T, T> function) {
+        if(from.compareTo(to)>0){
+            return;
+        }
+        this.from = from;
+        this.to = to;
+        if(!(from.equals(to))) {
+
+            do {
+                this.elementData.add(from);
+                from = function.apply(from);
+            }
+            while (!(from.equals(to)));
+            this.elementData.add(to);
+        }
+    }
 
-public class Range<T> implements Set<T> {
 
-    public int size() {
-        return 0;
+    public static Range<Integer> of(Integer from, Integer to) {
+        return new Range<>(from, to, intValue -> intValue + 1);
+    }
+
+    public static Range<Float> of(Float from, Float to) {
+        return new Range<>(from, to, floatValue -> floatValue + 0.1f);
     }
 
-    public boolean isEmpty() {
-        return false;
+    public static Range<Character> of(char from, char to, Function<Character, Character> characterCharacterFunction) {
```

Comment:
Add a generic **of** method.

Reply:
Static method can contains generic?

Reply:
Oh, you already have a constructor with al parameters, that's ok.

Reply:
You can have something like that static <T extends Number> Range<T> of(T e1, Character e2, Function<T, T> function) that contains generic, but constructor is fine.